### PR TITLE
DOC: Remove directives for some constants

### PR DESCRIPTION
**Pull request by takanori-pskq**
Related to #16217, and separated ffrom #16370. See https://github.com/numpy/numpy/pull/16370#discussion_r504556248 and https://github.com/numpy/numpy/pull/16370#discussion_r504555662.

### CODE REVIEW

File: doc/source/reference/c-api/dtype.rst [hunk 1]
```diff
@@ -221,24 +221,17 @@ Defines
 Max and min values for integers
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-.. c:var:: NPY_MAX_INT{bits}
-
-.. c:var:: NPY_MAX_UINT{bits}
-
-.. c:var:: NPY_MIN_INT{bits}
-
+``NPY_MAX_INT{bits}``, ``NPY_MAX_UINT{bits}``, ``NPY_MIN_INT{bits}``
```
Comment (eric-wieser) [inline doc/source/reference/c-api/dtype.rst:224]: I suppose the downside of this approach is that it doesn't create a link anchor?

File: doc/source/reference/c-api/dtype.rst [hunk 2]
```diff
@@ -221,24 +221,17 @@ Defines
 Max and min values for integers
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (eric-wieser) [inline doc/source/reference/c-api/dtype.rst:222]: https://16385-908607-gh.circle-artifacts.com/0/doc/build/html/reference/c-api/dtype.html#defines

File: doc/source/reference/c-api/ufunc.rst [hunk 3]
```diff
@@ -10,8 +10,7 @@ UFunc API
 Constants
 ---------
```
Comment (eric-wieser) [inline doc/source/reference/c-api/ufunc.rst:11]: https://16385-908607-gh.circle-artifacts.com/0/doc/build/html/reference/c-api/ufunc.html#constants

Comment (eric-wieser) [reply doc/source/reference/c-api/ufunc.rst:11]: Looks basically identical to https://numpy.org/devdocs/reference/c-api/ufunc.html anyway

**Review by eric-wieser — APPROVED** [review]

**Comment by mattip** [comment]
Thanks @takanori-pskq

**Pull request merged**